### ISSUE TITLE
php70u-7.0.22-2 does NOT install on CentOS 7

### ISSUE THREAD

**Issue by ralfbecker** [iuscommunity-pkg/php70u#14]
When I try a clean new install I get:
```
Error: Package: php70u-cli-7.0.22-2.ius.el7.x86_64 (ius)
           Requires: libcrypto.so.10(OPENSSL_1.0.2)(64bit)
```
Any idea?

Here's the complete log:
```
RalfsMac:php-ius ralf$ docker run -it --rm centos
Unable to find image 'centos:latest' locally
latest: Pulling from library/centos
Digest: sha256:26f74cefad82967f97f3eeeef88c1b6262f9b42bc96f2ad61d6f3fdf544759b8
Status: Downloaded newer image for centos:latest
[root@f12e66c8ca4a /]# cat /etc/*release
CentOS Linux release 7.3.1611 (Core)
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

CentOS Linux release 7.3.1611 (Core)
CentOS Linux release 7.3.1611 (Core)
[root@f12e66c8ca4a /]# rpm -i http://ftp.uni-stuttgart.de/epel/7/x86_64/e/epel-release-7-10.noarch.rpm
warning: /var/tmp/rpm-tmp.JgcIaf: Header V3 RSA/SHA256 Signature, key ID 352c64e5: NOKEY
[root@f12e66c8ca4a /]# rpm -i http://dl.iuscommunity.org/pub/ius/stable/Redhat/7/x86_64/ius-release-1.0-14.ius.el7.noarch.rpm
warning: /var/tmp/rpm-tmp.m5JTQz: Header V4 DSA/SHA1 Signature, key ID 9cd4953f: NOKEY
[root@f12e66c8ca4a /]# yum -y install php70u-cli php70u-json php70u-bcmath php70u-pdo php70u-opcache \
> php70u-xml php70u-ldap php70u-pecl-memcached php70u-mbstring php70u-tidy \
> php70u-mysqlnd php70u-process php70u-mcrypt php70u-gd php70u-pecl-igbinary \
> php70u-intl sudo && yum clean all
Loaded plugins: fastestmirror, ovl
base                                                                                                                                                                                                                                                  | 3.6 kB  00:00:00
epel/x86_64/metalink                                                                                                                                                                                                                                  |  19 kB  00:00:00
epel                                                                                                                                                                                                                                                  | 4.3 kB  00:00:00
extras                                                                                                                                                                                                                                                | 3.4 kB  00:00:00
ius                                                                                                                                                                                                                                                   | 2.3 kB  00:00:00
updates                                                                                                                                                                                                                                               | 3.4 kB  00:00:00
(1/8): base/7/x86_64/group_gz                                                                                                                                                                                                                         | 155 kB  00:00:01
epel/x86_64/updateinfo         FAILED
https://mirror.netcologne.de/fedora-epel/7/x86_64/repodata/28482b25bffc09658364a8e9e608fbf5f1755fbbfee59a627454f0741d2712cd-updateinfo.xml.bz2: [Errno 14] HTTPS Error 404 - Not Found                                                     ]  0.0 B/s |    0 B  --:--:-- ETA
Trying other mirror.
To address this issue please refer to the below knowledge base article

https://access.redhat.com/articles/1320623

If above article doesn't help to resolve this issue please create a bug on https://bugs.centos.org/

(2/8): extras/7/x86_64/primary_db                                                                                                                                                                                                                     | 191 kB  00:00:00
(3/8): ius/x86_64/primary_db                                                                                                                                                                                                                          | 211 kB  00:00:00
epel/x86_64/primary_db         FAILED                                                                                    2% [===                                                                                                           ]  0.0 B/s | 557 kB  --:--:-- ETA
http://mirror.wiuwiu.de/epel/7/x86_64/repodata/072b934b375efc1352ca5283c34c87e827f391742001ae5812b25ec7bcbab946-primary.sqlite.xz: [Errno 14] HTTP Error 404 - Not Found                                                                   ]  0.0 B/s | 557 kB  --:--:-- ETA
Trying other mirror.
(4/8): epel/x86_64/updateinfo                                                                                                                                                                                                                         | 810 kB  00:00:00
(5/8): epel/x86_64/group_gz                                                                                                                                                                                                                           | 170 kB  00:00:02
(6/8): base/7/x86_64/primary_db                                                                                                                                                                                                                       | 5.6 MB  00:00:03
(7/8): epel/x86_64/primary_db                                                                                                                                                                                                                         | 4.8 MB  00:00:02
(8/8): updates/7/x86_64/primary_db                                                                                                                                                                                                                    | 7.8 MB  00:00:04
Determining fastest mirrors
 * base: mirror.23media.de
 * epel: mirrors.n-ix.net
 * extras: mirror.rackspeed.de
 * ius: mirrors.ircam.fr
 * updates: mirror.daniel-jost.net
Resolving Dependencies
--> Running transaction check
---> Package php70u-bcmath.x86_64 0:7.0.22-2.ius.el7 will be installed
--> Processing Dependency: php-common(x86-64) = 7.0.22-2.ius.el7 for package: php70u-bcmath-7.0.22-2.ius.el7.x86_64
---> Package php70u-cli.x86_64 0:7.0.22-2.ius.el7 will be installed
--> Processing Dependency: libcrypto.so.10(OPENSSL_1.0.2)(64bit) for package: php70u-cli-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libedit.so.0()(64bit) for package: php70u-cli-7.0.22-2.ius.el7.x86_64
---> Package php70u-gd.x86_64 0:7.0.22-2.ius.el7 will be installed
--> Processing Dependency: libjpeg.so.62(LIBJPEG_6.2)(64bit) for package: php70u-gd-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libpng15.so.15(PNG15_0)(64bit) for package: php70u-gd-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libfreetype.so.6()(64bit) for package: php70u-gd-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libjpeg.so.62()(64bit) for package: php70u-gd-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libpng15.so.15()(64bit) for package: php70u-gd-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libwebp.so.4()(64bit) for package: php70u-gd-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libX11.so.6()(64bit) for package: php70u-gd-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libXpm.so.4()(64bit) for package: php70u-gd-7.0.22-2.ius.el7.x86_64
---> Package php70u-intl.x86_64 0:7.0.22-2.ius.el7 will be installed
--> Processing Dependency: libicudata.so.50()(64bit) for package: php70u-intl-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libicui18n.so.50()(64bit) for package: php70u-intl-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libicuio.so.50()(64bit) for package: php70u-intl-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libicuuc.so.50()(64bit) for package: php70u-intl-7.0.22-2.ius.el7.x86_64
---> Package php70u-json.x86_64 0:7.0.22-2.ius.el7 will be installed
---> Package php70u-ldap.x86_64 0:7.0.22-2.ius.el7 will be installed
---> Package php70u-mbstring.x86_64 0:7.0.22-2.ius.el7 will be installed
---> Package php70u-mcrypt.x86_64 0:7.0.22-2.ius.el7 will be installed
--> Processing Dependency: libltdl.so.7()(64bit) for package: php70u-mcrypt-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libmcrypt.so.4()(64bit) for package: php70u-mcrypt-7.0.22-2.ius.el7.x86_64
---> Package php70u-mysqlnd.x86_64 0:7.0.22-2.ius.el7 will be installed
---> Package php70u-opcache.x86_64 0:7.0.22-2.ius.el7 will be installed
---> Package php70u-pdo.x86_64 0:7.0.22-2.ius.el7 will be installed
---> Package php70u-pecl-igbinary.x86_64 0:2.0.1-1.ius.el7 will be installed
--> Processing Dependency: php70u-pear for package: php70u-pecl-igbinary-2.0.1-1.ius.el7.x86_64
--> Processing Dependency: php70u-pear for package: php70u-pecl-igbinary-2.0.1-1.ius.el7.x86_64
---> Package php70u-pecl-memcached.x86_64 0:3.0.3-1.ius.el7 will be installed
--> Processing Dependency: libevent-2.0.so.5()(64bit) for package: php70u-pecl-memcached-3.0.3-1.ius.el7.x86_64
--> Processing Dependency: libfastlz.so.0()(64bit) for package: php70u-pecl-memcached-3.0.3-1.ius.el7.x86_64
--> Processing Dependency: libmemcachedprotocol.so.0()(64bit) for package: php70u-pecl-memcached-3.0.3-1.ius.el7.x86_64
--> Processing Dependency: libmemcached.so.11()(64bit) for package: php70u-pecl-memcached-3.0.3-1.ius.el7.x86_64
--> Processing Dependency: libmemcachedutil.so.2()(64bit) for package: php70u-pecl-memcached-3.0.3-1.ius.el7.x86_64
---> Package php70u-process.x86_64 0:7.0.22-2.ius.el7 will be installed
---> Package php70u-tidy.x86_64 0:7.0.22-2.ius.el7 will be installed
--> Processing Dependency: libtidy.so.5()(64bit) for package: php70u-tidy-7.0.22-2.ius.el7.x86_64
---> Package php70u-xml.x86_64 0:7.0.22-2.ius.el7 will be installed
--> Processing Dependency: libxslt.so.1(LIBXML2_1.0.11)(64bit) for package: php70u-xml-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libxslt.so.1(LIBXML2_1.0.13)(64bit) for package: php70u-xml-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libxslt.so.1(LIBXML2_1.0.18)(64bit) for package: php70u-xml-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libxslt.so.1(LIBXML2_1.0.22)(64bit) for package: php70u-xml-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libxslt.so.1(LIBXML2_1.0.24)(64bit) for package: php70u-xml-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libexslt.so.0()(64bit) for package: php70u-xml-7.0.22-2.ius.el7.x86_64
--> Processing Dependency: libxslt.so.1()(64bit) for package: php70u-xml-7.0.22-2.ius.el7.x86_64
---> Package sudo.x86_64 0:1.8.6p7-23.el7_3 will be installed
--> Running transaction check
---> Package fastlz.x86_64 0:0.1.0-0.1.20070619svnrev12.el7 will be installed
---> Package freetype.x86_64 0:2.4.11-12.el7 will be installed
---> Package libX11.x86_64 0:1.6.3-3.el7 will be installed
--> Processing Dependency: libX11-common >= 1.6.3-3.el7 for package: libX11-1.6.3-3.el7.x86_64
--> Processing Dependency: libxcb.so.1()(64bit) for package: libX11-1.6.3-3.el7.x86_64
---> Package libXpm.x86_64 0:3.5.11-3.el7 will be installed
---> Package libedit.x86_64 0:3.0-12.20121213cvs.el7 will be installed
---> Package libevent.x86_64 0:2.0.21-4.el7 will be installed
---> Package libicu.x86_64 0:50.1.2-15.el7 will be installed
---> Package libjpeg-turbo.x86_64 0:1.2.90-5.el7 will be installed
---> Package libmcrypt.x86_64 0:2.5.8-13.el7 will be installed
---> Package libmemcached.x86_64 0:1.0.16-5.el7 will be installed
---> Package libpng.x86_64 2:1.5.13-7.el7_2 will be installed
---> Package libtidy.x86_64 0:5.4.0-1.el7 will be installed
---> Package libtool-ltdl.x86_64 0:2.4.2-22.el7_3 will be installed
---> Package libwebp.x86_64 0:0.3.0-3.el7 will be installed
---> Package libxslt.x86_64 0:1.1.28-5.el7 will be installed
---> Package php70u-cli.x86_64 0:7.0.22-2.ius.el7 will be installed
--> Processing Dependency: libcrypto.so.10(OPENSSL_1.0.2)(64bit) for package: php70u-cli-7.0.22-2.ius.el7.x86_64
---> Package php70u-common.x86_64 0:7.0.22-2.ius.el7 will be installed
---> Package php70u-pear.noarch 1:1.10.5-1.ius.el7 will be installed
--> Running transaction check
---> Package libX11-common.noarch 0:1.6.3-3.el7 will be installed
---> Package libxcb.x86_64 0:1.11-4.el7 will be installed
--> Processing Dependency: libXau.so.6()(64bit) for package: libxcb-1.11-4.el7.x86_64
---> Package php70u-cli.x86_64 0:7.0.22-2.ius.el7 will be installed
--> Processing Dependency: libcrypto.so.10(OPENSSL_1.0.2)(64bit) for package: php70u-cli-7.0.22-2.ius.el7.x86_64
--> Running transaction check
---> Package libXau.x86_64 0:1.0.8-2.1.el7 will be installed
---> Package php70u-cli.x86_64 0:7.0.22-2.ius.el7 will be installed
--> Processing Dependency: libcrypto.so.10(OPENSSL_1.0.2)(64bit) for package: php70u-cli-7.0.22-2.ius.el7.x86_64
--> Finished Dependency Resolution
Error: Package: php70u-cli-7.0.22-2.ius.el7.x86_64 (ius)
           Requires: libcrypto.so.10(OPENSSL_1.0.2)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
[root@f12e66c8ca4a /]#
```

**Comment by ralfbecker** [comment]
I tried the install again, this time with the CentOS 7 (not RHEL 7) repo and it worked.

I never used your CentOS repo before, always just the RHEL repo for both CentOS and RHEL.

Maybe the need to do so for PHP 7.0.22 need a more prominent place in the docu or release page (https://github.com/iuscommunity-pkg/php70u/releases/tag/7.0.22-2).

Ralf

**Comment by carlwgeorge** [comment]
Most of the time, the RHEL repo works on CentOS.  However, we are in that gap where RHEL has released a new point release (7.4) but CentOS has not yet completed their rebuild and are still using the previous point release (7.3).  The particular error you saw is because 7.4 introduces OpenSSL 1.0.2 (`Requires: libcrypto.so.10(OPENSSL_1.0.2)(64bit)`), which is not available in CentOS 7.3.

This scenario is not specific to PHP and so I feel that the current [installation page](https://ius.io/GettingStarted/#ius-release) is the most appropriate and prominent place for this information, and it's already there.

> IUS compiles separate packages for RHEL and CentOS.
